services/blockstorage/sync/block_sync.go

```diff
@@ -74,25 +88,27 @@ func (bs *BlockSync) syncLoop(ctx context.Context) {
 		// TODO add metrics
 		bs.currentState = bs.currentState.processState(ctx)
 	}
-
 }
 
-func (bs *BlockSync) HandleBlockCommitted() {
+func (bs *BlockSync) HandleBlockCommitted(ctx context.Context) {
+	ctx, _ = context.WithTimeout(ctx, bs.config.BlockSyncNoCommitInterval()/2)
```

Comment:
because as you suggested, this should not even be part of the code, the context should have a timeout as its coming from gossip, and we should add a timeout there on every call
until that happens, i wanted to add logic that will have it wait for less than the relevant action timer, so i /2 it